otherMethods.py

```diff
@@ -49,45 +49,55 @@ def download(cls, api, datasetName, entityIds, productName, output_dir):
         results_list = []
         for availableDownload in availableDownloads:
             url = availableDownload['url']
-            path = cls._download(url, output_dir)
+            if (productName.find('Bundle') == -1):
+                path = cls._download(url, output_dir)
+            else: 
+                path = cls._download(url, output_dir, bundle_dict={'name': url[url.find("product_id=") + len("product_id="):url.find("product_id=") + len("product_id=") + 38]})

```

Comment:
38 is the magic number (what does it mean?)

Reply:
When you request for the headers in the ```_download``` function, it returns the file name which you use.
Since for the BUNDLE files it wasnt working, I just used the url itself to get the name.
And from the total length of that string, I just substring the name out.
The length of the name of file here was 38 chars.